api/pkg/transformer/pipeline/table_join_op.go

```diff
@@ -0,0 +1,117 @@
+package pipeline
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/gojek/merlin/pkg/transformer/spec"
+	"github.com/gojek/merlin/pkg/transformer/types/table"
+)
+
+type TableJoinOp struct {
+	tableJoinSpec *spec.TableJoin
+}
+
+func NewTableJoinOp(tableJoinSpec *spec.TableJoin) Op {
+	return &TableJoinOp{
+		tableJoinSpec: tableJoinSpec,
+	}
+}
+
+func (t TableJoinOp) Execute(context context.Context, environment *Environment) error {
+	leftTable, err := getTable(environment, t.tableJoinSpec.LeftTable)
+	if err != nil {
+		return err
+	}
+
+	rightTable, err := getTable(environment, t.tableJoinSpec.RightTable)
+	if err != nil {
+		return err
+	}
+
+	var resultTable *table.Table
+	joinColumn := t.tableJoinSpec.On
+	switch t.tableJoinSpec.How {
+	case spec.JoinMethod_LEFT:
+		err := validateJoinColumn(leftTable, rightTable, joinColumn)
```

Comment:
why do we need this? if we call do `LeftJoin` it will also failed right if there is no overlapping column?

Reply:
It's to get a better error message, the error message thrown by gota is not user-friendly.